### PR TITLE
Fixing publishing issues and also adding branches for publishing docs

### PR DESCRIPTION
**Pull request by moizjv**
- Adding branches of format x.x.x

### CODE REVIEW

**Comment by jlipps** [comment]
:+1:

**Comment by jlipps** [comment]
merging since we have a release imminent, still interested in a @bootstraponline :+1: for feedback though

File: rake_lib/repo.rb [hunk 1]
```diff
@@ -123,8 +123,10 @@ def update_branches
           valid_branches << line.split('/').last.strip
         end
       end
-
+      # Adding branches in format x.x.x as per new release guidelines
+      @documentation_publish_branches = valid_branches.select{ |branch| branch[/^\d*\.\d*.\d*$/] }
       @branches = valid_branches.uniq.sort
+      puts "Branches for which documentation will be published: #{documentation_publish_branches.join(', ')}"
```
Comment (bootstraponline) [inline rake_lib/repo.rb]: this is bad ruby. you've changed the return value of this method to nil. it use to be `@branches`

also the puts should be using the branches attr, not the unsorted non-deduped documentation_publish_branches

File: rake_lib/helper.rb [hunk 2]
```diff
@@ -159,6 +159,8 @@ def update_docs is_master
       if is_master
         tags = %w[master]
       else
+        # Adding branches
+        tags = @appium_repo.documentation_publish_branches + tags
```
Comment (bootstraponline) [inline rake_lib/helper.rb]: branches are not tags. they are different concepts and handled separately in the code. I don't think mashing them together with tags is helpful.

**Pull request merged**